### ISSUE TITLE
[Accepted] SDL 0030 - Redesign iOS Logging

### ISSUE THREAD

**Issue by theresalech** [smartdevicelink/sdl_evolution#93]
Hello SDL community,

The review of "Redesign iOS Logging" begins now and runs through February 21, 2017. The proposal is available here:

https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0030-ios-logging-redesign.md

Reviews are an important part of the SDL evolution process. All reviews should be sent to the associated Github issue at:

https://github.com/smartdevicelink/sdl_evolution/issues/93

What goes into a review?

The goal of the review process is to improve the proposal under review through constructive criticism and, eventually, determine the direction of SDL. When writing your review, here are some questions you might want to answer in your review:

* Is the problem being addressed significant enough to warrant a change to SDL?
* Does this proposal fit well with the feel and direction of SDL?
* If you have used competitors with a similar feature, how do you feel that this proposal compares to those?
* How much effort did you put into your review? A glance, a quick reading, or an in-depth study?
Please state explicitly whether you believe that the proposal should be accepted into SDL.

More information about the SDL evolution process is available at

https://github.com/smartdevicelink/sdl_evolution/blob/master/process.md

Thank you,

Theresa Lech
Program Manager - Livio
theresa@livio.io

**Comment by kshala-ford** [comment]
This is quite a big proposal and a powerful feature for logging. Reminds me of CocoaLumberjack . Just spent more than an hour reading it. Here some thoughts:

- What files are going to be added from the SuperLogger repo? Including all the loggers/targets?
- What's the extra footprint for the binary? Developers complain about the size increase on iOS when integrating SDL framework.
- On iOS developers ask to get life logging to the computer. I always advised to use NSLogger. What's necessary to perform life logging with superlogger? What's your preferred target?

**Comment by joeljfischer** [comment]
@kshala-ford It is a big proposal, but since large sections of it have been implemented before, I don't think it will take that long to implement.

> What files are going to be added from the SuperLogger repo? Including all the loggers/targets?

Only what is described in the proposal will be added. So only the console, os_log, and file log targets. If more are desired, because it's extensible, others can build them and submit them back to the project if they'd like.

> What's the extra footprint for the binary? Developers complain about the size increase on iOS when integrating SDL framework.

Just the cost of adding a thousand or so lines of code. A few dozen kilobytes, perhaps? A hundred or two at the most? Considering that we're removing tens of thousands of LOC in 5.0, we're probably safe. I think the benefit is worth it anyway.

> On iOS developers ask to get live logging to the computer. I always advised to use NSLogger. What's necessary to perform live logging with superlogger? What's your preferred target?

With SuperLogger I had built a webserver target that allowed the device to serve a website on the local network that it could display logs on using WebSockets. That turned out to be a lot of work and was never finished. If live logging over the network is desired, anyone should be able to build a target (such as one to NSLogger, or a webserver logger, or a telnet logger) and submit it back to the project.
